src/main/scala/scala/concurrent/stm/ccstm/RetrySet.scala

```diff
@@ -22,7 +22,7 @@ private[ccstm] class RetrySet(val size: Int,
     val d = begin + timeoutNanos
     val deadline = if (d < 0) Long.MaxValue else d // handle arithmetic overflow
 
-    val timeoutExceeded = !attemptAwait(deadline)
```

Comment:
I wonder if you were supposed to do anythinig with the result `timeoutExceeded`. Not using the value could point to a bug or to an unnecessary call to `attemptAwait`?